provider/agent.go

```diff
@@ -38,6 +39,19 @@ func agentResource() *schema.Resource {
 					return diag.FromErr(err)
 				}
 			}
+
+			rawPlan := resourceData.GetRawPlan()
+			items := rawPlan.GetAttr("metadata").AsValueSlice()
+			itemKeys := map[string]struct{}{}
+			for _, item := range items {
+				key := valueAsString(item.GetAttr("key"))
+				_, exists := itemKeys[key]
+				if exists {
+					return errorAsDiagnostics(xerrors.Errorf("duplicate agent metadata key %q", key))
```

Comment:
Not sure why we have this function, it's basically a copy of a helper from the lib. So I would suggest this change:

```suggestion
					return diag.FromErr(xerrors.Errorf("duplicate agent metadata key %q", key))
```